restapi/src/main/java/io/stargate/web/docsapi/service/util/DocsApiUtils.java

```diff
@@ -122,6 +136,288 @@ private static String convertSingleArrayPath(String path) {
     return Optional.empty();
   }
 
+  private static Object convertToBackendBooleanValue(boolean value, boolean numericBooleans) {
+    if (numericBooleans) {
+      return value ? 1 : 0;
+    }
+    return value;
+  }
+
+  /**
+   * Transforms a JSON payload into a set of bind variables to send to Cassandra.
+   *
+   * @param surfer
+   * @param db
+   * @param path The path within the document that is being written to. If empty, writes to the root
+   *     of the document.
+   * @param key The name of the document that will be written
+   * @param payload a JSON object, or a URL-encoded form with the relevant data in it
+   * @param patching If this payload meant to be part of a PATCH request (this causes a small amount
+   *     of extra validation if true)
+   * @param isJson if the request had a content type of application/json, else it will be
+   *     interpreted as a URL encoded form
+   * @return The full bind variable list for the subsequent inserts, and all first-level keys, as an
+   *     ImmutablePair.
+   */
+  public static ImmutablePair<List<Object[]>, List<String>> shredPayload(
```

Comment:
this needs proper unit tests imo, the complexity of that path is so huge and we have no single unit tests, I would invest here in order to make sure this is good.. Also not sure if we should extract this to a separate shredding class, we are again getting huge classes..

In addition, we should start optimizing the code, we need to add more comments, split these huge methods into smaller ones, don't expose constants like `"dbl_value"` as plain text, etc..

Reply:
I agree - but I also think this would be a pretty sizeable chunk of work, and actually this feature is something that a customer was looking to have soon

In the interest of expedience for them, could I make a follow-up ticket to do this work and merge it as-is to get the feature in faster for them?

Reply:
I started work on this on a separate branch - will fast-follow after merging this, if that's alright

Reply:
perfect

Reply:
OK can we at least create a specific class for shredding and create a ticket to add tests and refactor please? 

Also can we make shred methods in non-static and make the class `@Inject DocsApiConfiguration docsApiConfiguration` so that we can expose methods that take defaults for the `docsApiConfiguration.getMaxDepth() & docsApiConfiguration.getMaxArrayLength()`? This way we don't need to have `DocsApiConfiguration` in the `ReactiveDocumentService` and any users of shred service as well?